### PR TITLE
Feature/us#42 organization cards

### PR DESCRIPTION
**Pull request by Art-mimiron**
#42

### CODE REVIEW

File: src/app/shell/main/result/hobby-groups-list/hobby-groups-list.component.html [hunk 1]
```diff
@@ -1 +1,22 @@
-<p>hobby-groups-list works!</p>
+<div class="cards" fxLayout="row wrap" fxLayoutGap='2%' fxLayoutAlign="flex-start" >
+  <div class="cards_wrapper" *ngFor='let card of cards'>
```
Comment (dmitryy90) [inline src/app/shell/main/result/hobby-groups-list/hobby-groups-list.component.html:2]: it is needed to be moved to card component and used as a web component which encapsulates card logic.

e g:

`
<card-list>
  <ng-container *ngFor="let card of (cards | async)">
    <card [card]="card"></card>
  </ng-container>
</card-list>`

File: src/app/shell/main/result/hobby-groups-list/hobby-groups-list.component.ts [hunk 2]
```diff
@@ -1,23 +1,29 @@
...
   styleUrls: ['./hobby-groups-list.component.scss']
 })
 export class HobbyGroupsListComponent implements OnInit {
-  stateCards: Observable<Object[]>;
-  public cards: Array<Object> = [];
+  /* stateCards: Observable<Object[]>; */
+  public cards: orgCard[];
 
-  constructor(private store: Store) {
-    this.stateCards = this.store.select(state => state.ShowData);
```
Comment (dmitryy90) [inline src/app/shell/main/result/hobby-groups-list/hobby-groups-list.component.ts:15]: try using @select decorator

File: src/app/shell/main/result/hobby-groups-list/hobby-groups-list.component.ts [hunk 3]
```diff
@@ -1,23 +1,29 @@
...
+  constructor(/* private store: Store */) {
+    /* this.stateCards = this.store.select(state => state.ShowData); */
     // To do: change "state.ShowData" when state will be configured
   }
   
   ngOnInit(): void {
-    this.stateCards.subscribe((data) => this.cards = data);
+    /* this.stateCards.subscribe((data) => this.cards = data); */
+    this.getCards().subscribe(cards => this.cards = cards);
```
Comment (dmitryy90) [inline src/app/shell/main/result/hobby-groups-list/hobby-groups-list.component.ts:22]: move to the sate

**Review by dmitryy90 — CHANGES_REQUESTED** [review]

File: src/app/shell/main/result/hobby-groups-list/hobby-groups-list.component.html (hunk 1 above)
Comment (Art-mimiron) [reply src/app/shell/main/result/hobby-groups-list/hobby-groups-list.component.html]: done

File: src/app/shell/main/result/hobby-groups-list/hobby-groups-list.component.ts (hunk 2 above)
Comment (Art-mimiron) [reply src/app/shell/main/result/hobby-groups-list/hobby-groups-list.component.ts:15]: that's was not my piece of code :)

File: src/app/shell/main/result/hobby-groups-list/hobby-groups-list.component.ts (hunk 3 above)
Comment (Art-mimiron) [reply src/app/shell/main/result/hobby-groups-list/hobby-groups-list.component.ts]: done

File: src/app/mock-org-cards.ts [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  city: string;
+  ownership: string;
+  price: number | boolean;
+  rate: string; //temp
+  ageFrom: number;
+  ageTo: number;
+  category: string[];
+}
+
+export const ORGCARDS: orgCard[] = [
```
Comment (dmitryy90) [inline src/app/mock-org-cards.ts:13]: do we still need this after mock was added?

File: src/app/shared/store/filter.state.ts [hunk 5]
```diff
@@ -23,12 +28,15 @@ export interface FilterStateModel {
     isRecruiting: true,
     ageFrom: null,
     ageTo: null,
-    categories: []
+    categories: [],
+    organizationCards: []
   }
```
Comment (dmitryy90) [inline src/app/shared/store/filter.state.ts:33]: @Selector()
  static organizationCards(state: FilterStateModel) {
    return state.organizationCards;
  }

File: src/app/shell/main/result/organization-cards-list/organization-cards-list.component.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+export class OrganizationCardsListComponent implements OnInit {
+  public cards: orgCard[];
+
+  @Select() filter$: Observable<FilterStateModel>
+
+  constructor(private store: Store) {}
+  
+  ngOnInit(): void {
+    this.store.dispatch(new getCards())
+    this.filter$.subscribe(filter => this.cards = filter.organizationCards)
```
Comment (dmitryy90) [inline src/app/shell/main/result/organization-cards-list/organization-cards-list.component.ts:23]: @Select(FilterState.organizationCards)
  organizationCards$: Observable<orgCard[]>;

**Review by dmitryy90 — CHANGES_REQUESTED** [review]

File: src/app/mock-org-cards.ts (hunk 4 above)
Comment (Art-mimiron) [reply src/app/mock-org-cards.ts:13]: removed

File: src/app/shared/store/filter.state.ts (hunk 5 above)
Comment (Art-mimiron) [reply src/app/shared/store/filter.state.ts:33]: +

File: src/app/shell/main/result/organization-cards-list/organization-cards-list.component.ts (hunk 6 above)
Comment (Art-mimiron) [reply src/app/shell/main/result/organization-cards-list/organization-cards-list.component.ts:23]: +

**Review by dmitryy90 — APPROVED** [review]

**Review by dmitryy90 — APPROVED** [review]

**Pull request merged**